### PR TITLE
CNAB Security 300

### PR DESCRIPTION
**Pull request by trishankatdatadog**
Major TODOs:
- [ ] Add a threat model to basically guide what 
- [x] Talk about signing only bundles, not images, so simplify security levels
- [ ] Discuss use cases to talk about what should be possible (e.g., we are doing this to support that use case)
- [x] Explain that it doesn't matter where you store bundles or metadata, or whether it's a think or thick bundle, because verification would exactly the same

Minor TODOs:
- [x] Replace links to ITEs

Sub-PRs:
- [x] [301 to discuss what metadata repositories may look like](https://github.com/deislabs/cnab-spec/pull/281)
- [x] [302 to discuss prescriptive signing workflows](https://github.com/deislabs/cnab-spec/pull/282)
- [x] [303 to discuss mandatory verification workflows](https://github.com/deislabs/cnab-spec/pull/280)
- [x] [304 to discuss known implementations](https://github.com/deislabs/cnab-spec/pull/288)

### CODE REVIEW

**Comment by trishankatdatadog** [comment]
Okay, I've decided to split everything up, and am using 300 as a roadmap. The pieces are as follows:

- 301 to discuss what metadata repositories may look like
- 302 to discuss prescriptive signing workflows
- 303 to discuss mandatory verification workflows

More to come. Stay tuned!

File: 300-CNAB-security.md [hunk 1]
```diff
@@ -5,168 +5,85 @@ weight: 300
...
 
-Layouts are well-defined in the In-Toto specification. However, a certain subset of In-Toto features are considered OPTIONAL in the present spec:
+ Signing and verifying [claims](400-claims.md) are out of the scope of this document.
 
-- Inspections: In-Toto provides a method for running arbitrary commands on a machine in order to verify that specific parts of a software supply chain are correct. This specification does not require support for inspections. Moreover, this specification discourages allowing bundles to run external commands even for inspection purposes.
+## Roadmap
 
-### Verifying Layouts
+The basic idea is to TUF as a transport protocol that distributes several files in a compromise-resilient manner:
```
Comment (vdice) [inline 300-CNAB-security.md]: Should this read "... is to _use_ TUF as a..."?

File: 300-CNAB-security.md [hunk 2]
```diff
@@ -5,168 +5,85 @@ weight: 300
...
 
-The In-Toto specification defines the procedure for verifying individual links against the layout. This section adds CNAB-specific context to the more general specification.
+* The root of trust for all bundles and images, as well as TUF and in-toto metadata.
+* The software supply chains defined using in-toto.
+* The public keys used to verify these supply chains.
 
-If any part of a layout cannot be satisfied, the client SHOULD generate an error and MUST NOT continue installing the bundle.
+While images and bundles are published on registries, TUF and in-toto metadata about images and bundles are published on [_metadata repositories_](301-CNAB-metadata-repositories.md). Metadata repositories are not conceptually, but not necessarily physically, distinct from [CNAB](200-CNAB-registries.md) or [OCI](https://github.com/opencontainers/distribution-spec/blob/master/spec.md) registries. Furthermore, the key management models on metadata repositories are largely prescriptive. For these reasons and more, metadata repositories out of the scope of this document.
```
Comment (vdice) [inline 300-CNAB-security.md]: Should the second sentence read, "Metadata repositories are conceptually..."? (Dropping the `not`?)

Comment (vdice) [reply 300-CNAB-security.md]: Should the last sentence read, "For these reasons and more, metadata repositories _are_ out of the scope of this document."? (Missing `are`?)

File: 300-CNAB-security.md [hunk 3]
```diff
@@ -5,168 +5,85 @@ weight: 300
...
 
-If a link does not match a step in the layout, the link SHOULD be ignored. But if multiple links apply to the same step of a layout, and at least one can be verified, that layout stem SHOULD be considered verified.
+Unlike specified otherwise, metadata are TUF and in-toto metadata, which may be produced using the suggested [signing workflows](302-signing-workflows.md).
```
Comment (vdice) [inline 300-CNAB-security.md]: s/Unlike/Unless, right?

File: 300-CNAB-security.md [hunk 4]
```diff
@@ -5,168 +5,85 @@ weight: 300
...
-<signature-scheme>$<signed-digest>$<gzipped-tar-data>
-```
+Level 1a: images are signed using TUF, but bundles are unsigned.
+   * Pros: easiest level of security to achieve, e.g. using Docker Content Trust (metadata repository) and Notary (client).
+   * Cons: only partly compromise-resilient. If a [CNAB registry](200-CNAB-registries.md) is compromised, a bundle could be easily replaced with malicious one.
 
-Where `<signature-scheme>` is to be replaced by one of the following schemes, `<signature-digest>` is to be replaced by the signature, and `<gzipped-tar-data>` is to be replaced by the raw gzipped tar data. The `$` characters are literal.
+Level 1b: bundles are signed using TUF, but images are unsigned.
+   * Pros: easiest level of security to achieve, e.g. using Docker Content Trust (metadata repository) and Notary (client).
+   * Cons: only partly compromise-resilient. If an [OCI registry](https://github.com/opencontainers/distribution-spec/blob/master/spec.md) is compromised, a bundler runtime may be vulnerable to accidentally bundling malicious images.
```
Comment (vdice) [inline 300-CNAB-security.md]: s/bundler runtime/bundle runtime, right?

**Comment by trishankatdatadog** [comment]
Thanks for your feedback @vdice!

**Comment by trishankatdatadog** [comment]
Could @vdice or @chris-crone please 👀, ✅, and merge? Thanks!

**Pull request merged**